core/shared/src/main/scala/matryoshka/patterns/ListF.scala

```diff
@@ -52,6 +52,14 @@ object ListF {
     case NilF()      => Seq.empty
   } (s => s.headOption.fold[ListF[A, Seq[A]]](NilF())(ConsF(_, s.tail)))
 
+  def takeUpTo[N, T, A](implicit N: Recursive.Aux[N, Option], T: Recursive.Aux[T, ListF[A, ?]]): Coalgebra[ListF[A, ?], (N, T)] =
+    pair => pair._1.project.fold[ListF[A, (N, T)]](NilF())(p => pair._2.project.map((p, _)))
+
+  def find[A](cond: A => Boolean): Algebra[ListF[A, ?], Option[A]] = {
```

Comment:
I’m generally pretty wary about wrapping things up with their folds. I do it in `matryoshka.instances.fixedpoint` to illustrate how we can implement existing interfaces, but one of the big benefits of recursion schemes is the compositionality, and hiding the algebras (as we used to do in Quasar, but have moved away from) removes that. That’s why (for the most part) the algebras used _by_ `matryoshka.instances.fixedpoint` are defined in `matryoshka.patterns` – you generally want to try to work with patterns instead of instances.